sdk/cosmos/azure_data_cosmos/src/clients/container_client.rs

```diff
@@ -666,23 +683,38 @@ impl ContainerClient {
     /// ```
     ///
     /// See [`PartitionKey`](crate::PartitionKey) for more information on how to specify a partition key, and [`Query`] for more information on how to specify a query.
-    pub fn query_items<T: DeserializeOwned + Send>(
+    pub fn query_items<T: DeserializeOwned + 'static>(
         &self,
         query: impl Into<Query>,
-        partition_key: impl Into<QueryPartitionStrategy>,
+        partition_key: impl Into<PartitionKey>,
         options: Option<QueryOptions<'_>>,
     ) -> azure_core::Result<FeedPager<T>> {
-        let options = options.unwrap_or_default();
-        let url = self.pipeline.url(&self.items_link);
-        let mut base_request = Request::new(url, Method::Post);
-        let QueryPartitionStrategy::SinglePartition(partition_key) = partition_key.into();
-        base_request.insert_headers(&partition_key)?;
+        #[allow(unused_mut)] // This is used in the #[cfg(feature = "query_engine")] block
```

Comment:
Maybe this instead, then? Less fragile to accidents - almost like `#[expect]` you wanted back when (and I'd still like to use).
```suggestion
        #[cfg_attr(not(feature = "query_engine"), allow(unused_mut))]
```

Reply:
Yep, I have done that in other places, I like it better. I'll do that!